sources/packages/backend/apps/db-migrations/src/sql/Types/Add-institution-restriction-type.sql

```diff
@@ -0,0 +1,16 @@
+-- Postgres allows adding new types to an enum but it causes issues when the new types added are
+-- used in another query in the same transaction, hence the team decision was to recreate the enums
+-- types when a new item must be added following the same approach already used for rollbacks.
+CREATE TYPE sims.restriction_types_to_be_updated AS ENUM ('Provincial', 'Federal', 'Institution');
+
+-- Update the dependent column to start using the new enum with the expected values.
+ALTER TABLE
+    sims.restrictions
+ALTER COLUMN
+    restriction_type TYPE sims.restriction_types_to_be_updated USING (restriction_type :: TEXT) :: sims.restriction_types_to_be_updated;
+
+-- Remove the entire enum that is no longer being used.
+DROP TYPE sims.restriction_types;
+
+-- Rename the enum to what it was in the beginning.
+ALTER TYPE sims.restriction_types_to_be_updated RENAME TO restriction_types;
```

Comment:
im guessing this reflects back to the table column, so no need to alter the column again right?

Reply:
Great question. Seems like Postgres will handle this.

Reply:
Yes, it will reflect back to the column.
This file can be used as a pattern to execute these operations. We have quite a few doing the same 😉